AirportTicketBooking/Flights/FlightRepository.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Globalization;
+using System.IO;
+using AirportTicketBooking.Services;
+using CsvHelper;
+
+namespace AirportTicketBooking.Flights
+{
+    public class FlightRepository
+    {
+        private readonly CsvReader _csvReader;
+        private readonly CSVIOService _csvioService;
+
+        public FlightRepository(CsvReader csvReader, CSVIOService csvioService)
+        {
+            _csvReader = csvReader;
+            _csvioService = csvioService;
+        }
+        
+        public void InsertFlights(string filePath, ReadFromCsvFile readFromCsvFile, FileServices fileServices, Validation validation)
+        {
+            if (!filePath.Contains(".csv"))
+            {
+                throw new Exception("The file type is ont supported!");
+            }
+
+            if (File.Exists(filePath))
+            {
+                readFromCsvFile.RegisterFlightsData(filePath, fileServices, validation);
+            }
+            else
+            {
+                throw new NullReferenceException("The file you are trying to access doesn't exist!");
+            }
```

Comment:
NullReferenceException is not sutable for this case, search for FileNotFoundException.

Reply:
Thanks for the note.